draft-looker-oauth-attested-key-based-client-authentication.md

```diff
@@ -94,34 +94,51 @@ The following diagram depicts the conceptual interactions.
+The flow starts with the Authorization Server communicating its requirements for the client authentication to the client, made available in its metadata. A Client instance that wants to request an access token from this Authorization Server needs to obtain a Client Attestation JWT, for example, from its client backend first.
 
-This specification only defines the format of the client assertion that a client instance uses to authenticate in its interactions with an authorization server (indicated in step 6), which is comprised of two key parts:
+Therefore, the client generates a key (Client Instance Key) and (platform specific) attestations to prove its authenticity, integrity and security to the client backend (step 1). The Client instance sends this data to the client backend in request for a Client Attestation JWT (step 2). If the Client Backend successfully validates the Client Instance Key and further data, it generates a signed Client Attestation JWT (step 3). As the Client Attestation JWT is cryptographically bound to the Client Instance Key generated by the client, the attestation is bound to this particular client instance. The client backend responds to the client's request by sending the Client Attestation JWT (step 4).
 
-1. A client key attestation - produced by the client backend.
-2. A client key attestation proof of possession (PoP) - produced by the client instance.
+The client can proceed now and generate a Client Attestation Proof of Possession (PoP) for the Client Instance Key (step 5). Lastly, the client sends both the Client Attestation JWT and the Client Attestation PoP with the Token Request to the Authorization Server. The Authorization Server validates the client attestation and thus authenticates the client. The Authorization Server may continue to issue sender-constrained access tokens using {{DPOP}}.
+
+Note that the protocol for steps 2 and 4 and how the client instance authenticates with the client backend is out of scope of this specification.
+
+This specification defines the format of the Client Attestation that a client instance uses to authenticate in its interactions with an authorization server, which is comprised of two key parts:
+
+1. A Client Attestation JWT- produced by the client backend.
+2. A Client Attestation Proof of Possession (PoP) - produced by the client instance.
 
 # Conventions and Definitions
 
 {::boilerplate bcp14-tagged}
 
+# Terminology
+
+Client Attestation JWT
+   A JSON Web Token (JWT) generated by the client backend that authenticates the client instance.
+
+Client Attestation Proof of Possession (PoP)
+   A Proof of Possession generated by the client instance for the key that is used in the Client Attestation JWT.
+
+Client Instance Key
+   A cryptographic, asymmetric key generated by the client instance and proven to the client backend. The public key is contained in the Client Attestation JWT and is used to sign the Client Attestation Proof of Possession.
+
```

Comment:
I approve this PR with this in mind

Client Instance, as represented in the Figure 1, should be a defined term as well